baseplate.go

```diff
@@ -78,16 +78,16 @@ type Server interface {
 }
 
 // Serve runs the given Server until it is given an external shutdown signal
-// using runtimebp.HandleShutdown to handle the signal and shut down the
-// server gracefully.  Returns the (possibly nil) error returned by "Close" or
-// context.DeadlineExceeded if it times out.
+// using runtimebp.HandleShutdown to handle the signal and shut down the server
+// and any provided closers gracefully.  Returns the (possibly nil) error
+// returned by "Close", or context.DeadlineExceeded if it times out.
 //
 // If a StopTimeout is configure, Serve will wait for that duration for the
 // server to stop before timing out and returning to force a shutdown.
 //
 // This is the recommended way to run a Baseplate Server rather than calling
 // server.Start/Stop directly.
-func Serve(ctx context.Context, server Server) error {
+func Serve(ctx context.Context, server Server, closers ...io.Closer) error {
```

Comment:
In addition to some extra documentation, maybe we should name these closers something more descriptive?  Like "onShutdown" or maybe even "preShutdown"?